api/server/router/plugin/plugin_routes.go

```diff
@@ -116,20 +116,77 @@ func (pr *pluginRouter) pullPlugin(ctx context.Context, w http.ResponseWriter, r
 	}
 
 	metaHeaders, authConfig := parseHeaders(r.Header)
+	ref, tag, err := parseRemoteRef(r.FormValue("remote"))
+	if err != nil {
+		return err
+	}
+
+	name, err := getName(ref, tag, vars["name"])
+	if err != nil {
+		return err
+	}
+	w.Header().Set("Docker-Plugin-Name", name)
```

Comment:
@tiborvass I just copied this from the existing code: https://github.com/docker/docker/pull/29414/files#diff-3869b4c752615bcebd087e8cf5eee874L154